core/src/main/scala/io/delta/tables/DeltaMergeBuilder.scala

```diff
@@ -199,13 +201,39 @@ class DeltaMergeBuilder private(
     DeltaMergeNotMatchedActionBuilder(this, Some(condition))
   }
 
+  /**
+   * Explain the merge operation based on the built matched and not matched actions.
+   *
+   * @since 1.2.0
+   */
+  def explain(): Unit = {
+    val sparkSession = targetTable.toDF.sparkSession
+    val mergeIntoCommand = getMergeIntoCommand(sparkSession)
+
+    // only simple is needed to describe the merge into command
+    val explain = ExplainCommand(mergeIntoCommand, mode = ExplainMode.fromString("simple"))
+    val plan = SparkSession.active.sessionState.executePlan(explain).executedPlan.executeCollect()
+      .map(_.getString(0))
+      .mkString("\n")
+
+    // scalastyle:off println
+    println(plan)
```

Comment:
is this what SQL command `EXPLAIN MERGE INTO... ` will be print? Only the raw logical plan?

Reply:
So this should be what the EXPLAIN MERGE INTO prints with the relations/references resolved.

```scala
val output = sql(
        s"""
          |EXPLAIN MERGE INTO delta.`$tempPath`
          |USING source
          |ON key1 = key2
          |WHEN MATCHED AND TRUE THEN DELETE
          |WHEN MATCHED THEN UPDATE SET key1 = key2, value1 = value1
          |WHEN NOT MATCHED THEN INSERT (key1, value1) VALUES (key2, value2)""".stripMargin)
      output.show(truncate = false)
```

That returns this:

```
== Physical Plan ==\nExecute MergeIntoCommand\n   +- MergeIntoCommand SubqueryAlias source, SubqueryAlias spark_catalog.delta.`/private/var/folders/gy/dy_2wchs6wz7223r9nzckf980000gq/T/spark-859ffb5e-0efc-469b-9913-e2a7a7801c96`, Delta[version=0, ... :/private/var/folders/gy/dy_2wchs6wz7223r9nzckf980000gq/T/spark-859ffb5e-0efc-469b-9913-e2a7a7801c96], (key1#482 = key2#473), [Delete [condition: true], Update [actions: [`key1` = key2#473, `value1` = value1#483]]], [Insert [actions: [`key1` = key2#473, `value1` = value2#474]]], StructType(StructField(key1,IntegerType,true), StructField(value1,IntegerType,true))\n\n|
```

The `.explain()` command:

```
== Physical Plan ==
Execute MergeIntoCommand
   +- MergeIntoCommand Project [_1#468 AS key2#473, _2#469 AS value2#474], Relation [key1#477,value1#478] parquet, Delta[version=0, ... :/private/var/folders/gy/dy_2wchs6wz7223r9nzckf980000gq/T/spark-7609aad4-527b-409f-8a78-61175255890a], (key1#477 = key2#473), [Delete [condition: true], Update [actions: [`key1` = key2#473, `value1` = value1#478]]], [Insert [actions: [`key1` = key2#473, `value1` = value2#474]]], StructType(StructField(key1,IntegerType,true), StructField(value1,IntegerType,true))
```

It is almost the same except for the source relation.